### ISSUE TITLE
Using TypeAlias containing TypeVar resulting in unknown in TypeAlias

### ISSUE THREAD

**Issue by dephiros** [microsoft/pyright#6715]
**Describe the bug**
When using `TypeAlias` that includes a `TypeVar`, the `TypeVar` results in unknown type
Using the TypeVar directly does not result in the error

**Code or Screenshots**

This is a simplified example from [django-stubs `_DisplayT` type](https://github.com/typeddjango/django-stubs/blob/master/django-stubs/contrib/admin/options.pyi#L136) that produces the same error
```python
from typing import TypeVar, Callable, TypeAlias

_ModelT = TypeVar("_ModelT", bound=str)
_DisplayT: TypeAlias = str | Callable[[_ModelT], str | bool]


def generic_func(arg: _ModelT, arg1: _ModelT) -> _DisplayT:
    return arg


def generic_func1(arg: _ModelT, arg1: _ModelT) -> str | Callable[[_ModelT], str | bool]:
    return arg

```
<img width="1222" alt="Screenshot 2023-12-12 at 08 23 12" src="https://github.com/microsoft/pyright/assets/2289753/8bf22cb2-0120-4e07-b184-aa0b3a4cb0d7">

**VS Code extension or command-line**
*pyright*: 1.1.339

**Comment by erictraut** [comment]
Pyright's behavior is correct here, so this isn't a bug.

You're creating a generic type alias that has one type parameter. That means any time you use the type alias, you should provide a single type argument. If you don't provide a type argument, type checkers will assume that you mean `Any`. This is the same as with generic types. For example, if you use `list` in a type annotation, a static type checker will assume that you mean `list[Any]`.

You can fix the bug in your code by replacing `_DisplayT` with `_DisplayT[_ModelT]` in the `generic_func` function signature.

If you enable the `reportMissingTypeArgument` diagnostic rule in pyright, it will tell you when you have omitted a type argument for a generic class or type alias.

If you're using Python 3.12, you can switch to the new type parameter syntax (defined in [PEP 695](https://peps.python.org/pep-0695/)), which is less confusing than the older syntax. Here's how your code looks with the new syntax:

```python
from typing import Callable

type Display[T: str] = str | Callable[[T], str | bool]

def generic_func[T: str](arg: T, arg1: T) -> Display[T]:
    return arg
```
